api/v1alpha1/storage_types.go

```diff
@@ -105,17 +106,23 @@ type StorageData struct {
 	Capacity int64 `json:"capacity"`
 
 	// Status is the overall status of the storage
-	// +kubebuilder:validation:Enum=Starting;Ready;Disabled;NotPresent;Offline;Failed
+	// +kubebuilder:validation:Enum:=Starting;Ready;Disabled;NotPresent;Offline;Failed
 	Status string `json:"status,omitempty"`
 }
 
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="State",type="string",JSONPath=".spec.state",description="State of the storage resource"
+//+kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.status",description="Status of the storage resource"
+//+kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+
 // Storage is the Schema for the storages API
-// +kubebuilder:object:root=true
 type Storage struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Data StorageData `json:"data,omitempty"`
+	Spec   StorageSpec   `json:"spec,omitempty"`
```

Comment:
You can remove the omitempty.  You don't actually want to allow one of these resources to be created without a spec section, right?

Reply:
Maybe re-work `config/samples/dws_v1alpha1_storage.yaml` as well.

Reply:
I am just following precedence. ClientMount, SystemConfiguration, and Servers all allow for an empty spec. Should we move all those resources forward at the same time?

Reply:
I wonder what would break if we did that?  Maybe a test here or there would have to be fixed?

Reply:
Does spec with omitempty allow for `r.Status().Update()` to occur? All this originates from operator-sdk, so I'm sure they had a reason and I'm not sure I want to play with fire here.

Reply:
Doesn't affect that.  This originates from kubebuilder, which operator-sdk uses.  They default to omitempty because they're also providing a sample in config/samples, and they want the sample to apply out-of-the-box but they don't know how to construct a valid spec for you...

In my kubebuilder experiments I've never hit a problem when removing that omitempty.

Reply:
OK - thanks for the helpful explanations. I've made the change to Storage.Spec; will tackle other objects in a future PR